### ISSUE TITLE
Implementing AcmeProxy provider which doesn't support getting or updating records

### ISSUE THREAD

**Issue by mdbraber** [libdns/libdns#6]
I'm planning on using `libdns` for [acmeproxy](https://github.com/mdbraber/acmeproxy), which "proxies" TXT record Present/Cleanup requests at a central location to prevent spreading API credentials all over my network. I'm positive this will work fine using `libdns` internally to append/delete/set those records.

The other side of this to provide an Acmeproxy implementation of libdns for tools that use libdns for ACME validation, e.g. Caddy. How can I best implement this as `GetRecords` is not implemented in acmeproxy (just Present and Cleanup) and `AppendRecords` is equal to `SetRecords`. Would it be okay just to implement it with GetRecords returning `nil`?

**Comment by mholt** [comment]
So, currently Caddy's use of libdns packages is in transition. Right now Caddy has a little wrapper inside it that implements an acme solver/provider interface. After I'm done rewriting the underlying acme package, that interface will be different, and it probably won't look like the Present/Cleanup you know today.

The only answer I can give you right now is, "Don't implement acmeproxy as a libdns provider, because it is not a DNS provider." The proper solution would be to implement the more generic/generalized challenge solver interface, which is still TBD.

Basically, I don't have an answer for you quite yet; I will need to figure it out after the rewrite.

**Comment by mdbraber** [comment]
I understand - my only objective would be that there's a way to implement acmeproxy as a dns01 challenge provider in Caddy. I'll keep an eye on how to do that in the future and will close this for now.

**Comment by mholt** [comment]
Sounds good, please keep in touch so we can get it right! Thanks
